src/java/voldemort/store/rocksdb/RocksDbStorageConfiguration.java

```diff
@@ -34,13 +37,39 @@
 
     private Map<String, RocksDbStorageEngine> stores = new HashMap<String, RocksDbStorageEngine>();
 
+    // RocksDB Configuration Options
+    private final CompactionStyle compactionStyle;
+    private final CompressionType compressionType;
+    private final int maxBackgroundCompactions;
+    private final int maxBackgroundFlushes;
+    private final long maxBytesForLevelBase;
+    private final int maxWriteBufferNumber;
+    private final int statsDumpPeriodSec;
+    private final long targetFileSizeBase;
+    private final int targetFileSizeMultiplier;
+    private final long writeBufferSize;
+
     public RocksDbStorageConfiguration(VoldemortConfig config) {
         /**
          * - TODO 1. number of default locks need to debated. This default is
          * same as that of Krati's. 2. Later add the property to VoldemortConfig
          */
         this.voldemortconfig = config;
-        this.lockStripes = config.getAllProps().getInt("rocksdb.lock.stripes", 50);
+        Props props = config.getAllProps();
+
+        this.lockStripes = props.getInt("rocksdb.lock.stripes", 50);
+
+        // TODO: Validate the default mandatory options
+        compactionStyle = CompactionStyle.valueOf(props.getString("rocksdb.options.compactionStyle", CompactionStyle.UNIVERSAL.toString()));
```

Comment:
Hmm...

I thought I had given a comment here earlier, but apparently it got lost in the mists of the interwebs. Anyway, it went something along the lines of:

Sorry for another last minute nitpick, but can we make this config extraction part of ```VoldemortConfig```? It would be ideal if all configs, and their default values, were defined in there only.

Thanks for your work! It's very nice to see someone pushing on the RocksDB front (:

-F